client/wfnews-war/src/main/angular/src/app/components/wildfires-list/wildfires-list.component.ts

```diff
@@ -81,20 +81,38 @@ export class WildFiresListComponent extends CollectionComponent implements OnCha
           return;
       }
 
-      if(val.length > 2) {
-          this.filteredOptions= [];
-
-          this.placeData.searchAddresses(val).then(function(results){
-              if(results) {
-
-                  results.forEach((result) => {
-                      let address = self.getFullAddress(result);
-                      result.address = address.trim();
-                  });
-
-                  self.filteredOptions = results;
+      if (val.length > 2) {
+        this.filteredOptions = [];
+        let address = null;
+        let trimmedAddress = null;
+        let valueLength = null;
+        let valueMatch = null;
+        this.placeData.searchAddresses(val).then(function (results) {
+
+          if (results) {
+            results.forEach((result) => {
+              address = self.getFullAddress(result);
+              result.address = address.trim();
+              trimmedAddress = result.address;
+              valueLength = val.length;
+              if (trimmedAddress != null) valueMatch = trimmedAddress.substring(0, valueLength);
+
+              if (address != null && valueLength != null && valueMatch != null &&
+                (val.toUpperCase() === address.toUpperCase() || val.toUpperCase() === valueMatch.toUpperCase())) {
+                  var index = results.indexOf(result);
```

Comment:
Replace the var with let or const (as above, I assume const here as it doesnt change)